### ISSUE TITLE
AddApplicationParts Example references deprecated package.

### ISSUE THREAD

**Issue by ConnerPhillis** [dotnet/AspNetCore.Docs#29806]
The example provided for AddApplicationParts references code that is deprecated for (at least) .NET 7.
Microsoft.AspNetCore.App has been deprecated.

[see nuget package here](https://www.nuget.org/packages/Microsoft.AspNetCore.App/)

how are we expected to add additional parts to an application now without making the dependent assembly itself an ASP.NET Core project?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 2784a724-e940-4c14-3b50-b063f78c2ee5
* Version Independent ID: 41175c3a-8cb5-856b-a7e0-931edb6394db
* Content: [Share controllers, views, Razor Pages and more with Application Parts in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/mvc/advanced/app-parts?view=aspnetcore-7.0)
* Content Source: [aspnetcore/mvc/advanced/app-parts.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/advanced/app-parts.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @ConnerPhillis ... The package is depreciated for standalone use, but the namespace/assembly is still valid with full support. The API has been rolled into the shared framework. For more information, see https://learn.microsoft.com/aspnet/core/fundamentals/metapackage-app.

**Comment by ConnerPhillis** [comment]
Okay - but when using AddApplicationParts, wouldn't that library that contains the assemblies that would be imported be considered a standalone usage, so deprecated?

Sorry if I'm not understanding, thanks for your quick reply!

**Comment by ConnerPhillis** [comment]
Wait - I understand now. That's a framework reference, not a package reference.

That worked, assemblies are now resolving. Thank you!

**Comment by guardrex** [comment]
Yes, that's right ... framework reference. The API is fully supported that way.

**Comment by ConnerPhillis** [comment]
Any chance the sample would be updated in those docs, in that case?

**Comment by guardrex** [comment]
Possibly ... this article maps to Rick, so I'll re-open the issue. He'll be along in a bit to take a look ... he's in another timezone :smile:.

**Comment by guardrex** [comment]
Where were you talking about making the update tho because I see it in one of the samples correctly as a framework reference ...

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/advanced/app-parts/3.0sample1/MySharedApp/MySharedApp.csproj#L16

**Comment by guardrex** [comment]
The rest of the examples seem to be 2.x era, so they're fine the way that they are.

**Comment by ConnerPhillis** [comment]
This is the link referenced in the doc

https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/mvc/advanced/app-parts/sample2

I was looking at

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/advanced/app-parts/sample2/Plugin/Plugin.csproj

**Comment by guardrex** [comment]
That's ok as sample code given that it's 2.x era.

I guess Rick will need to see how that's linked into the topic. Sounds like the versioning of that linked code might be off.

**Comment by ConnerPhillis** [comment]
Looks like if I would have navigated up in that repo directory I would have seen it, but I didn't. Maybe the link just needs to go to the parent directory?

**Comment by guardrex** [comment]
It looks like there's a bad cross-link at Line 72 ... it links to the 2.x code but in the >=3.0 doc version.

**Comment by guardrex** [comment]
It's a little bit complicated. That part of the >=3.0 content doesn't seem to be part of the 3.0 sample app. The reference was carried forward instead. Rick will take a look when he gets here and will see how to handle this.

Anyway ... the main issue is sorted out. I'm glad you have your bits working now.

Leave this issue open. He'll be along in a bit to take a look.
